front/src/modules/simulationResult/components/ChartHelpers/enableInteractivity.tsx

```diff
@@ -0,0 +1,384 @@
+    .attr('x', (d) => newX(d[xAxisStart]!))
+    .attr('y', (d) => newY(d[yAxisStart]!) - (newY(d[yAxisEnd]!) - newY(d[yAxisStart]!)) * -1)
+    .attr('width', (d) => newX(d[xAxisEnd]!) - newX(d[xAxisStart]!))
+    .attr('height', (d) => (newY(d[yAxisEnd]!) - newY(d[yAxisStart]!)) * -1);
+
+  // CLARA: in SpeedSpaceChart & SpaceCurveSlopes
+  const newAreaData = rotate
+    ? d3
+        .area<AreaBlock>()
+        .y((d) => newY(d.position))
+        .x0((d) => newX(d.value0))
+        .x1((d) => newX(d.value1))
+        .curve(isSpaceTimeChart(keyValues) ? d3.curveStepAfter : d3.curveLinear)
+    : d3
+        .area<AreaBlock>()
+        .x((d) => newX(d.position))
+        .y0((d) => newY(d.value0))
+        .y1((d) => newY(d.value1))
+        .curve(isSpaceTimeChart(keyValues) ? d3.curveStepAfter : d3.curveLinear);
+
+  chart.drawZone.selectAll<SVGPathElement, AreaBlock[]>('.area').attr('d', newAreaData);
+
+  // OPERATIONNAL POINTS
+  if (rotate) {
+    chart.drawZone
+      .selectAll('#get-operationalPointsZone .op-line')
+      .attr('x1', (d) => newX(d as number))
+      .attr('x2', (d) => newX(d as number));
+    chart.drawZone
+      .selectAll('#get-operationalPointsZone .op-text')
+      .attr('x', (d) => newX(d as number));
+    chart.drawZone
+      .selectAll('#gev-operationalPointsZone .op-line')
+      .attr('y1', (d) => newY(d as number))
+      .attr('y2', (d) => newY(d as number));
+    chart.drawZone
+      .selectAll('#gev-operationalPointsZone .op-text')
+      .attr('y', (d) => newY(d as number));
+  } else {
+    chart.drawZone
+      .selectAll('#get-operationalPointsZone .op-line')
+      .attr('y1', (d) => newY(d as number))
+      .attr('y2', (d) => newY(d as number));
+    chart.drawZone
+      .selectAll('#get-operationalPointsZone .op-text')
+      .attr('y', (d) => newY(d as number));
+    chart.drawZone
+      .selectAll('#gev-operationalPointsZone .op-line')
+      .attr('x1', (d) => newX(d as number))
+      .attr('x2', (d) => newX(d as number));
+    chart.drawZone
+      .selectAll('#gev-operationalPointsZone .op-text')
+      .attr('x', (d) => newX(d as number));
+  }
+
+  chart.drawZone.selectAll('.curve-label').attr('transform', event.transform.toString());
+
+  // CLARA: removed conflictsPoints as it seems to never be created
+  return { newX, newY };
+};
+
+// Factorizes func to update VerticalLine on 3 charts: SpaceTime, SpeedSpaceChart, SpaceCurvesSlopes
+export const traceVerticalLine = (
+  chart: Chart | undefined,
+  keyValues: ChartAxes,
+  positionValues: PositionsSpeedTimes<Date>,
+  rotate: boolean,
+  timePosition: Date
+) => {
+  if (chart !== undefined) {
+    displayGuide(chart, 1);
+    if (rotate) {
+      chart.svg
+        .selectAll('#horizontal-line')
+        .attr(
+          'y1',
+          chart.y(
+            !isSpaceTimeChart(keyValues) && positionValues.speed
+              ? positionValues.speed.position
+              : timePosition
```

Comment:
This piece of code is used 4 times, we can create a const linePosition and use it instead

Reply:
absolutely !